src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsReader.java

```diff
@@ -101,6 +117,21 @@ public ByteVectorValues getByteVectorValues(final String field) throws IOExcepti
      */
     @Override
     public void search(String field, float[] target, KnnCollector knnCollector, Bits acceptDocs) throws IOException {
+        if (knnCollector instanceof QuantizationConfigKNNCollector) {
```

Comment:
lets put a TODO here suggesting that this a hack, where we are using KNNCollector to init the QS.

Reply:
nit: Would have preferred type casting the reader instead of this. not super urgent